backends/ref/ceed-ref-operator.c

```diff
@@ -1026,6 +1026,38 @@ static int CeedOperatorLinearAssembleAddPointBlockDiagonal_Ref(CeedOperator op,
   }
 }
 
+//------------------------------------------------------------------------------
+// Build Mass, Laplacian matrices
+//------------------------------------------------------------------------------
+CeedPragmaOptimizeOff
+static int CeedBuildMassLaplace(const CeedScalar *interp_1d,
+                                const CeedScalar *grad_1d,
+                                const CeedScalar *q_weight_1d, CeedInt P_1d,
+                                CeedInt Q_1d, CeedInt dim,
+                                CeedScalar *mass, CeedScalar *laplace) {
+
+  for (CeedInt i=0; i<P_1d; i++)
+    for (CeedInt j=0; j<P_1d; j++) {
+      CeedScalar volatile sum = 0.0;
+      for (CeedInt k=0; k<Q_1d; k++)
+        sum += interp_1d[k*P_1d+i]*q_weight_1d[k]*interp_1d[k*P_1d+j];
+      mass[i+j*P_1d] = sum;
+    }
+  // -- Laplacian
+  for (CeedInt i=0; i<P_1d; i++)
+    for (CeedInt j=0; j<P_1d; j++) {
+      CeedScalar volatile sum = 0.0;
```

Comment:
`volatile` is almost never the answer, but is frequently part of magical lore. I'd prefer that it be removed.
```suggestion
      CeedScalar sum = 0.0;
```